src/app/hooks.ts

```diff
@@ -0,0 +1,13 @@
+// This file serves as a central hub for re-exporting pre-typed Redux hooks.
+// These imports are restricted elsewhere to ensure consistent
+// usage of typed hooks throughout the application.
+// We disable the ESLint rule here because this is the designated place
+// for importing and re-exporting the typed versions of hooks.
+// eslint-disable @typescript-eslint/no-restricted-imports
+import { useDispatch, useSelector } from "react-redux";
+import type { RootState, AppDispatch } from "./store";
+
+// Use throughout your app instead of plain `useDispatch` and `useSelector`
+// eslint-disable @typescript-eslint/no-restricted-imports
+export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
+export const useAppSelector = useSelector.withTypes<RootState>();
```

Comment:
The code is good, but please remove the commented lines as these constants are just for exporting purpose and we will most likely never come back to this file.